### DISCUSSION TITLE
Should ELK be the default layout engine?

### DISCUSSION THREAD

**Discussion by alixander** [terrastruct/d2#990] [Polls]
Currently we use dagre, but with the recent improvements to ELK, I find myself choosing it more than dagre nowadays.

**Comment by bo-ku-ra** [comment]
i think it's a litlle bit early
 - elk tends to be overlaped connector texts. (solved?)
 - sketch mode is more spoiled than dagre

**Comment by HariSekhon** [comment]
No, I've tested it on most of my diagrams here and they come out better with dagre:

https://github.com/HariSekhon/Diagrams-as-Code

This also shouldn't be done before https://github.com/terrastruct/d2/issues/410 and ample time and warnings across several versions for users to hardcode their layout engine within their d2 script files as per https://github.com/terrastruct/d2/issues/1290.

If anything, it perhaps should never be done because backwards compatibility is a bad user experience for anybody who has invested a lot of time in a technology.

It'd be better to just make it a prominent setting recommendation in tutorials and docs and let users do it to avoid breaking existing things.

I personally generate all my D2 diagrams via GitHub Actions CI/CD in that repo above and would be very unimpressed if one day they all got messed up by some change under the hood of d2.
